### PR TITLE
Use RobotElementNames

### PR DESCRIPTION
**Pull request by muscatd75**

### CODE REVIEW

File: src/main/cpp/Robot.cpp [hunk 1]
```diff
@@ -121,7 +121,7 @@ void Robot::RobotPeriodic()
         feedback->UpdateFeedback();
     }
 
-    auto pigeon = RobotConfigMgr::GetInstance()->GetCurrentConfig()->GetPigeon(CanSensorUsage::CANSENSOR_USAGE::PIGEON_ROBOT_CENTER);
+    auto pigeon = RobotConfigMgr::GetInstance()->GetCurrentConfig()->GetPigeon(RobotElementNames::PIGEON_USAGE::PIGEON_ROBOT_CENTER);
```
Comment (witcpalek) [inline src/main/cpp/Robot.cpp:124]: I think we can remove this logging from Robot.cpp (swerve chassis already does this).

The cleaner we can keep Robot.cpp the better.

File: src/main/cpp/chassis/swerve/SwerveChassis.cpp [hunk 2]
```diff
@@ -123,7 +123,7 @@ SwerveChassis::SwerveChassis(
                                m_maxAngularSpeed(maxAngularSpeed),
                                m_maxAcceleration(maxAcceleration),               // Not used at the moment
                                m_maxAngularAcceleration(maxAngularAcceleration), // Not used at the moment
-                               m_pigeon(RobotConfigMgr::GetInstance()->GetCurrentConfig()->GetPigeon(CanSensorUsage::CANSENSOR_USAGE::PIGEON_ROBOT_CENTER)),
+                               m_pigeon(RobotConfigMgr::GetInstance()->GetCurrentConfig()->GetPigeon(RobotElementNames::PIGEON_USAGE::PIGEON_ROBOT_CENTER)),
```
Comment (witcpalek) [inline src/main/cpp/chassis/swerve/SwerveChassis.cpp:126]: this is temporary, but ok

File: src/main/cpp/chassis/swerve/driveStates/TrajectoryDrivePathPlanner.cpp [hunk 3]
```diff
@@ -99,7 +99,7 @@ std::array<frc::SwerveModuleState, 4> TrajectoryDrivePathPlanner::UpdateSwerveMo
         Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, "Trajectory Drive Path Planner", "HolonomicRotation (Degs)", m_desiredState.targetHolonomicRotation.Degrees().to<double>());
         Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, "Trajectory Drive Path Planner", "Omega (Rads Per Sec)", refChassisSpeeds.omega.to<double>());
         Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, "Trajectory Drive Path Planner", "Yaw Odometry (Degs)", m_chassis->GetPose().Rotation().Degrees().to<double>());
-        Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, "Trajectory Drive Path Planner", "Yaw Pigeon (Degs)", RobotConfigMgr::GetInstance()->GetCurrentConfig()->GetPigeon(CanSensorUsage::CANSENSOR_USAGE::PIGEON_ROBOT_CENTER)->GetYaw().to<double>());
+        Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, "Trajectory Drive Path Planner", "Yaw Pigeon (Degs)", RobotConfigMgr::GetInstance()->GetCurrentConfig()->GetPigeon(RobotElementNames::PIGEON_USAGE::PIGEON_ROBOT_CENTER)->GetYaw().to<double>());
```
Comment (witcpalek) [inline src/main/cpp/chassis/swerve/driveStates/TrajectoryDrivePathPlanner.cpp:102]: This is temporary, but ok

File: src/main/cpp/hw/factories/DigitalInputFactory.cpp [hunk 4]
```diff
@@ -22,7 +22,7 @@
 // Team 302 includes
 #include "hw/DragonDigitalInput.h"
 #include "hw/factories/DigitalInputFactory.h"
-#include "configs/usages/DigitalInputUsage.h"
+#include "configs/RobotElementNames.h"
```
Comment (witcpalek) [inline src/main/cpp/hw/factories/DigitalInputFactory.cpp:25]: Shouldn't this factory go away?  Isn't the code generator the thing that is create DigitalInputs now?

File: src/main/cpp/hw/factories/DigitalInputFactory.h [hunk 5]
```diff
@@ -20,7 +20,7 @@
 #include <string>
```
Comment (witcpalek) [inline src/main/cpp/hw/factories/DigitalInputFactory.h:20]: Shouldn't this file get deleted?

File: src/main/cpp/hw/factories/DragonServoFactory.cpp [hunk 6]
```diff
@@ -20,7 +20,7 @@
```
Comment (witcpalek) [inline src/main/cpp/hw/factories/DragonServoFactory.cpp:20]: Shouldn't this file get deleted?

File: src/main/cpp/hw/factories/DragonServoFactory.h [hunk 7]
```diff
@@ -22,7 +22,7 @@
 #include "units/angle.h"
```
Comment (witcpalek) [inline src/main/cpp/hw/factories/DragonServoFactory.h:22]: Shouldn't this file get deleted?

**Review by witcpalek — CHANGES_REQUESTED** [review]

File: src/main/cpp/Robot.cpp (hunk 1 above)
Comment (muscatd75) [reply src/main/cpp/Robot.cpp:124]: removed logging and committed

File: src/main/cpp/chassis/swerve/SwerveChassis.cpp (hunk 2 above)
Comment (muscatd75) [reply src/main/cpp/chassis/swerve/SwerveChassis.cpp:126]: ok

File: src/main/cpp/hw/factories/DigitalInputFactory.cpp (hunk 4 above)
Comment (muscatd75) [reply src/main/cpp/hw/factories/DigitalInputFactory.cpp:25]: Added a story

File: src/main/cpp/hw/factories/DigitalInputFactory.h (hunk 5 above)
Comment (muscatd75) [reply src/main/cpp/hw/factories/DigitalInputFactory.h:20]: added a story

File: src/main/cpp/hw/factories/DragonServoFactory.cpp (hunk 6 above)
Comment (muscatd75) [reply src/main/cpp/hw/factories/DragonServoFactory.cpp:20]: added a story

File: src/main/cpp/hw/factories/DragonServoFactory.h (hunk 7 above)
Comment (muscatd75) [reply src/main/cpp/hw/factories/DragonServoFactory.h:22]: added a story

**Review by witcpalek — APPROVED** [review]

**Pull request merged**